### DISCUSSION TITLE
Cooldown reset/blocking problem

### DISCUSSION THREAD

**Discussion by kozaka-tv** [Th0rstenf/Rocksniffer-to-Streamer.bot-extension#103] [Ideas]
From "Make cooldownPeriod configurable" #80 I removed the side effect problem, what we should solve in a different issue but discuss maybe here first.

(...but maybe this is only because there is a bug in develop right now because of change "Moved timer check one level to make it more robust #98 ")

The side effect is, that if scene changes during the timeout period, the timeout time will reset, and will be started to measure again.

Let's say cooldown is set to 6 seconds. Auto switch is set to 3 seconds (or in OBS streamer changes the scenese "manually" and relative fast). After each 3 seconds, the timeout will be set to 0. And "Is in cooldown" will be almost every time true. Or changed rapidly from false to true in one process step.

**Comment by Th0rstenf** [comment]
Please verify again when the current pull requests are in. There is a solid chance it should be fixed then :)

**Comment by Th0rstenf** [comment]
> Let's say cooldown is set to 6 seconds. Auto switch is set to 3 seconds (or in OBS streamer changes the scenese "manually" and relative fast). After each 3 seconds, the timeout will be set to 0. And "Is in cooldown" will be almost every time true. Or changed rapidly from false to true in one process step.

That would only mean the user misconfigured it, so minimum period would be cool down in practice. 
Why would one configure a cool down longer longer than the intended scene switch? This is really not something that we need to work around if. 
Just stating it in Readme or the SB comment respectively is enough.

**Reply by kozaka-tv** [reply]
I am just thinking...maybe this cooldown is good, in case we are switching from main states like:
- menü
- song start
- song end
- pause

And maybe also: 
- enter solo
- leave solo

And if in song, cooldown should be not used. I mean, it is better, if do not control this during a song, should irrelevant for us.
I know cooldown is not = with delay, but if the user want to have switched scene during a song, faster than we define, could be that the user will be unhappy.

I do not like if any the applications could give us freedom in certain things, but not gives, and restrict the creator this way. For example, timer in SB minimum is 1 second. Or logging an error is not possible.

**Reply by Th0rstenf** [reply]
Well, the original implementation did not really consider multiple scenes for songs anyway. (I and most people I know just use nested scenes, so there is technically one song scene, that still has a lot of variation.

What if we apply this cooldown only to switches outside of songscenes?

In that case we could potentially remove it as configuration option again, because then the user won't be limited during a song (aside from the SB execution frequency of course), and anything else is not a relevant usecase (certainly no one wants to jump in / out of songs at this rate)

Could this be the best solution?

**Reply by kozaka-tv** [reply]
Yes, I think also, would be better if we apply this cooldown only to switches outside of songScenes.

An interesting topic is, regarding multiple song scenes in groups, that, is the auto switch working at all if song scenes are grouped? I will create a discussion regarding this.

**Comment by kozaka-tv** [comment]
Issue #109 created.
